File: articles/gratitude-positive-psychology/fortran/gratitude_mechanism_index.f90
program gratitude_mechanism_index
  implicit none
  real :: gratitude_score, life_satisfaction, perceived_support, resilience_score
  real :: stress_load, depressive_symptoms, reflection_depth, gratitude_expression
  real :: intervention_fit, relationship_quality, social_trust
  real :: appreciative_orientation, relational_support, net_wellbeing

  gratitude_score = 7.2
  life_satisfaction = 7.1
  perceived_support = 7.3
  resilience_score = 6.9
  stress_load = 3.6
  depressive_symptoms = 3.8
  reflection_depth = 7.0
  gratitude_expression = 7.1
  intervention_fit = 7.2
  relationship_quality = 7.4
  social_trust = 7.1

  appreciative_orientation = (gratitude_score + perceived_support + reflection_depth + &
      gratitude_expression + relationship_quality + social_trust + intervention_fit - stress_load) / 8.0

  relational_support = (perceived_support + relationship_quality + social_trust + gratitude_expression) / 4.0

  net_wellbeing = life_satisfaction + gratitude_score + perceived_support + resilience_score + &
      reflection_depth + gratitude_expression + relationship_quality + social_trust - &
      depressive_symptoms - stress_load

  print *, "Appreciative orientation index:", appreciative_orientation
  print *, "Relational support index:", relational_support
  print *, "Net wellbeing index:", net_wellbeing
end program gratitude_mechanism_index

File: articles/gratitude-positive-psychology/go/gratitude_mechanism_index.go
package main

import "fmt"

type GratitudeIndicators struct {
	GratitudeScore      float64
	LifeSatisfaction    float64
	PerceivedSupport    float64
	ResilienceScore     float64
	StressLoad          float64
	DepressiveSymptoms  float64
	ReflectionDepth     float64
	GratitudeExpression float64
	InterventionFit     float64
	RelationshipQuality float64
	SocialTrust         float64
}

func AppreciativeOrientationIndex(x GratitudeIndicators) float64 {
	return (x.GratitudeScore +
		x.PerceivedSupport +
		x.ReflectionDepth +
		x.GratitudeExpression +
		x.RelationshipQuality +
		x.SocialTrust +
		x.InterventionFit -
		x.StressLoad) / 8.0
}

func RelationalSupportIndex(x GratitudeIndicators) float64 {
	return (x.PerceivedSupport +
		x.RelationshipQuality +
		x.SocialTrust +
		x.GratitudeExpression) / 4.0
}

func NetWellbeingIndex(x GratitudeIndicators) float64 {
	return x.LifeSatisfaction +
		x.GratitudeScore +
		x.PerceivedSupport +
		x.ResilienceScore +
		x.ReflectionDepth +
		x.GratitudeExpression +
		x.RelationshipQuality +
		x.SocialTrust -
		x.DepressiveSymptoms -
		x.StressLoad
}

func main() {
	example := GratitudeIndicators{7.2, 7.1, 7.3, 6.9, 3.6, 3.8, 7.0, 7.1, 7.2, 7.4, 7.1}
	fmt.Printf("Appreciative orientation index: %.3f\n", AppreciativeOrientationIndex(example))
	fmt.Printf("Relational support index: %.3f\n", RelationalSupportIndex(example))
	fmt.Printf("Net wellbeing index: %.3f\n", NetWellbeingIndex(example))
}
